src/libraries/System.Private.CoreLib/src/System/ArraySegment.cs

```diff
@@ -19,6 +20,8 @@ namespace System
     // (ie, users could assign a new value to the old location).
     [Serializable]
     [TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
+    [DebuggerDisplay("{DebuggerToString(),nq}")]
+    [DebuggerTypeProxy(typeof(ArraySegment<>.ArraySegmentDebugView))]
```

Comment:
I believe that we should simply mimic what `List<T>` does:

https://github.com/dotnet/runtime/blob/568ed4125d0ae62c0d82d8f3703854fcc04bf32d/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L17C1-L18

`ArraySegment<T>` implements `IList<T>` which extends `ICollection<T>` so it should work OOTB without any other code changes being required


```suggestion
    [DebuggerTypeProxy(typeof(ICollectionDebugView<>))]
    [DebuggerDisplay("Count = {Count}")]
```